lib/AST/DiagnosticEngine.cpp

```diff
@@ -1429,17 +1429,11 @@ DiagnosticEngine::diagnosticInfoForDiagnostic(const Diagnostic &diagnostic,
     }
   }
 
-  llvm::StringRef format;
-  if (includeDiagnosticName) {
-    format =
-        diagnosticStringWithNameFor(diagnostic.getID(), groupID,
-                                    getPrintDiagnosticNamesMode());
-  } else {
-    format = diagnosticStringFor(diagnostic.getID());
-  }
+  auto formatString =
+      getFormatStringForDiagnostic(diagnostic, includeDiagnosticName);
 
   return DiagnosticInfo(diagnostic.getID(), loc, toDiagnosticKind(behavior),
-                        format, diagnostic.getArgs(), Category,
+                        formatString, diagnostic.getArgs(), Category,
```

Comment:
The "category" is what drives diagnostic groups now (`-print-diagnostic-groups` is on its way to being removed). Are we getting the category and educational notes from the wrapped diagnostic?

Reply:
No, the only thing we currently get from the wrapped diag is the group. This change is just for the ID we display for `-debug-diagnostic-names`. I can do the category & notes next.